src/main/java/edu/cornell/kfs/fp/batch/service/impl/AuxiliaryVoucherDocumentGenerator.java

```diff
@@ -0,0 +1,234 @@
+package edu.cornell.kfs.fp.batch.service.impl;
+
+import java.sql.Date;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.apache.commons.lang3.StringUtils;
+import org.joda.time.DateTime;
+import org.kuali.kfs.coa.businessobject.AccountingPeriod;
+import org.kuali.kfs.coa.service.AccountingPeriodService;
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.kuali.kfs.fp.document.AuxiliaryVoucherDocument;
+import org.kuali.kfs.fp.document.validation.impl.AuxiliaryVoucherDocumentRuleConstants;
+import org.kuali.kfs.krad.bo.AdHocRoutePerson;
+import org.kuali.kfs.krad.bo.Note;
+import org.kuali.kfs.krad.exception.ValidationException;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.businessobject.AccountingLine;
+import org.kuali.kfs.sys.service.UniversityDateService;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+import org.kuali.rice.core.api.parameter.ParameterEvaluatorService;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.kew.api.WorkflowDocument;
+
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentAccountingLine;
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentEntry;
+
+public class AuxiliaryVoucherDocumentGenerator
+        extends AccountingDocumentGeneratorBase<AuxiliaryVoucherDocument> {
+
+    protected AccountingPeriodService accountingPeriodService;
+    protected UniversityDateService universityDateService;
+    protected ParameterEvaluatorService parameterEvaluatorService;
+    protected ParameterService parameterService;
+    protected DateTimeService dateTimeService;
+
+    public AuxiliaryVoucherDocumentGenerator() {
+        super();
+    }
+
+    public AuxiliaryVoucherDocumentGenerator(
+            Supplier<Note> emptyNoteGenerator, Supplier<AdHocRoutePerson> emptyAdHocRoutePersonGenerator) {
+        super(emptyNoteGenerator, emptyAdHocRoutePersonGenerator);
+    }
+
+    @Override
+    public Class<? extends AuxiliaryVoucherDocument> getDocumentClass() {
+        return AuxiliaryVoucherDocument.class;
+    }
+
+    @Override
+    protected <A extends AccountingLine> A buildAccountingLine(
+            Class<A> accountingLineClass, String documentNumber, AccountingXmlDocumentAccountingLine xmlLine) {
+        A accountingLine = super.buildAccountingLine(accountingLineClass, documentNumber, xmlLine);
+        validateAndPopulateAccountingLineWithDebitCreditAmounts(accountingLine, xmlLine);
+        return accountingLine;
+    }
+
+    protected <A extends AccountingLine> void validateAndPopulateAccountingLineWithDebitCreditAmounts(
+            A accountingLine, AccountingXmlDocumentAccountingLine xmlLine) {
+        if (hasAmountBeenSpecified(xmlLine.getAmount())) {
+            throw new ValidationException(
+                    "AV accounting line cannot use the 'amount' element to specify an amount; use the relevant debit/credit amount element instead");
```

Comment:
I'd suggesting moving this and other validation exception messages to properties

Reply:
Ok, I'll get these messages moved to properties accordingly.